packages/ns-dpi/files/dpi

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+
+#
+# Copyright (C) 2023 Nethesis S.r.l.
+# SPDX-License-Identifier: GPL-2.0-only
+#
+
+graceful_stop() {
+  echo "Stopping service..."
+  cleanup_tables
+  exit 0
+}
+
+cleanup_tables() {
+  echo "Cleaning up tables..."
+  nft delete table inet dpi
+}
+
+load_configuration() {
+  echo "Loading configuration..."
+
+  # To enable CT labels, create an iptables rule (similar rules on nftables do not enable CT labels).
+  for c in INPUT OUTPUT FORWARD; do
+    if ! iptables -t mangle -C $c -m connlabel --label dummy -m comment --comment "Init CT labels" 2>/dev/null; then
+        iptables -t mangle -A $c -m connlabel --label dummy -m comment --comment "Init CT labels";
+    fi
+  done
+
+  # Setup CT label for nftables which uses a different path than iptables
+  ln -sf /etc/xtables/connlabel.conf /etc
+
+  # Generate netify flow actions config file
+  /usr/sbin/dpi-config
+  # reload netifyd daemon
+  /etc/init.d/netifyd reload
+
+  # in case of filtering enabled, add nft tables, otherwise remove them
+  if [ "$(uci get dpi.config.enabled)" = "1" ]; then
+    /usr/sbin/dpi-nft | nft -f -
+  else
+    cleanup_tables
+  fi
+}
+
+# trapping cleanup event
+trap graceful_stop TERM
+
+# trap SIGHUP to reload configuration
+trap load_configuration HUP
+
+# generate configuration
+load_configuration
+
+# most sophisticated program ever wrote
+while true; do
+   sleep 1 &
+   wait $!
+done
```

Comment:
Can you try with a simpler implementation like:
```suggestion
sleep infinity
```

Reply:
Tried this exact approach, however it seems that `sleep infinity` inhibits signaling traps in this specific environment, no clue why.

Reply:
Okay, this introduces a brand new quirk!

In case of a restart of `dpi`, the sleep infinity sits in the background indefinitely :partying_face:

`procd` is weird.

Reply:
Tried 
```bash
sleep infinite &
wait $!
```

no luck

Reply:
I could just
```bash
tail -f /dev/null
```

Reply:
Ok for now, but I need to do some tests on a real machine

Reply:
So, we will go with my code or a `tail`?

Reply:
I'd prefer tail for now

Reply:
Tail makes the process to ignore SIGs too. I assume there's an issue with `procd`.

What you like me to do? I see within the system various `sleep *`, so I assume is a known issue.

Alternatively, I can put a restart instead of a reload, but sill `procd` will kill the process and report it in the log.